test/core/end2end/tests/resource_quota_server.cc

```diff
@@ -66,13 +68,19 @@ TEST_P(ResourceQuotaTest, ResourceQuota) {
   auto requests = MakeVec([](int) { return RandomSlice(128 * 1024); });
   auto server_calls =
       MakeVec([this](int i) { return RequestCall(kServerRecvBaseTag + i); });
-  std::vector<IncomingMetadata> server_metadata(kNumCalls);
-  std::vector<IncomingStatusOnClient> server_status(kNumCalls);
-  std::vector<IncomingMessage> client_message(kNumCalls);
-  std::vector<IncomingCloseOnServer> client_close(kNumCalls);
+  IncomingMetadata server_metadata[kNumCalls];
+  IncomingStatusOnClient server_status[kNumCalls];
+  IncomingMessage client_message[kNumCalls];
+  IncomingCloseOnServer client_close[kNumCalls];
+  enum class SeenServerCall {
+    kNotSeen = 0,
+    kSeenWithSuccess,
+    kSeenWithFailure
+  };
+  SeenServerCall seen_server_call[kNumCalls] = {SeenServerCall::kNotSeen};
```

Comment:
Probably worth a comment saying that this initializes the whole array, not just the first element.